File: NQT/NumberSystem/NumberToWord.java
package NumberSystem;

public class NumberToWord {
    private static final String[] units = {
            "", "One", "Two", "Three", "Four", "Five", "Six", "Seven", "Eight", "Nine", "Ten",
            "Eleven", "Twelve", "Thirteen", "Fourteen", "Fifteen", "Sixteen", "Seventeen", "Eighteen", "Nineteen"
    };
    private static final String[] tens = {
            "", "", "Twenty", "Thirty", "Forty", "Fifty", "Sixty", "Seventy", "Eighty", "Ninety"
    };
    private static final String[] thousands = {
            "", "Thousand", "Million"
    };

    public static void main(String[] args) {
        //number can be upto millions, you can change this value to test with different numbers
        int number = 12345; // You can change this value to test with different numbers
        String wordValue = numberToWords(number);
        System.out.println("The word representation of number " + number + " is:  " + wordValue);
    }

    private static String numberToWords(int number) {
        if (number == 0) {
            return "Zero";
        }
        StringBuilder wordValue = new StringBuilder();
        int thousandCounter = 0;

        while (number > 0) {
            if (number % 1000 != 0) {
                wordValue.insert(0, convertLessThanThousand(number % 1000) + " " + thousands[thousandCounter] + " ");
            }
            number /= 1000;
            thousandCounter++;
        }
        return wordValue.toString().trim();
    }

    private static String convertLessThanThousand(int number) {
        StringBuilder word = new StringBuilder();
        if (number >= 100) {
            word.append(units[number / 100]).append(" Hundred ");
            number %= 100;
        }
        if (number >= 20) {
            word.append(tens[number / 10]).append(" ");
            number %= 10;
        }
        if (number > 0) {
            word.append(units[number]).append(" ");
        }
        return word.toString().trim();
    }

}

File: NQT/NumberSystem/Python/number_to_word.py
# 1234 -> one thousand two hundred thirty four

def number_to_words(n):
    if n == 0:
        return "zero"

    def one(num):
        switcher = {
            1: "one",
            2: "two",
            3: "three",
            4: "four",
            5: "five",
            6: "six",
            7: "seven",
            8: "eight",
            9: "nine"
        }
        return switcher.get(num)

    def two_less_20(num):
        switcher = {
            10: "ten",
            11: "eleven",
            12: "twelve",
            13: "thirteen",
            14: "fourteen",
            15: "fifteen",
            16: "sixteen",
            17: "seventeen",
            18: "eighteen",
            19: "nineteen"
        }
        return switcher.get(num)

    def ten(num):
        switcher = {
            2: "twenty",
            3: "thirty",
            4: "forty",
            5: "fifty",
            6: "sixty",
            7: "seventy",
            8: "eighty",
            9: "ninety"
        }
        return switcher.get(num)

    def two(num):
        if num == 0:
            return ""
        elif num < 10:
            return one(num)
        elif num < 20:
            return two_less_20(num)
        else:
            tenner = num // 10
            rest = num % 10
            return ten(tenner) + ("-" + one(rest) if rest != 0 else "")

    def three(num):
        hundred = num // 100
        rest = num % 100
        if hundred and rest:
            return one(hundred) + " hundred and " + two(rest)
        elif not hundred and rest:
            return two(rest)
        elif hundred and not rest:
            return one(hundred) + " hundred"

    billion = n // 1000000000
    million = (n - billion * 1000000000) // 1000000
    thousand = (n - billion * 1000000000 - million * 1000000) // 1000
    remainder = n % 1000